src/btree/bt_sync.c

```diff
@@ -607,7 +607,9 @@ __wt_sync_file(WT_SESSION_IMPL *session, WT_CACHE_OP syncop)
              * Once the transaction has given up it's snapshot it is no longer safe to reconcile
              * pages. That happens prior to the final metadata checkpoint.
              */
-            if (F_ISSET(walk, WT_REF_FLAG_LEAF) && page->read_gen == WT_READGEN_WONT_NEED &&
+            if (F_ISSET(walk, WT_REF_FLAG_LEAF) &&
+              (page->read_gen == WT_READGEN_WONT_NEED ||
+                FLD_ISSET(conn->timing_stress_flags, WT_TIMING_STRESS_CHECKPOINT_EVICT_PAGE)) &&
```

Comment:
Is it worth making this more subtle - and evicting a random proportion of pages, rather than every page?

Reply:
I agree, although I would prefer merging this in with a WT reproducer first, and then create a follow-up ticket on deciding on any further improvements regarding this flag. If we evict a random portion of pages the chances of hitting the bug will be alot lower and even possibly go away.

Reply:
sgtm